csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -99,19 +100,49 @@ public static string ComputeFileHash(string filePath)
             return hex.ToString();
         }
 
-        private static async Task DownloadFileAsync(string address, string filename)
+        private static async Task DownloadFileAsync(string address, string filename, HttpClient httpClient, CancellationToken token)
         {
-            using var httpClient = new HttpClient();
-            using var contentStream = await httpClient.GetStreamAsync(address);
+            using var contentStream = await httpClient.GetStreamAsync(address, token);
             using var stream = new FileStream(filename, FileMode.Create, FileAccess.Write, FileShare.None, 4096, true);
-            await contentStream.CopyToAsync(stream);
+            await contentStream.CopyToAsync(stream, CancellationToken.None);
```

Comment:
Do we want to pass in the `token` to the cancellation token?

Reply:
I don't know. I guess we could include the copying in the specified timeout too.

Reply:
I would assume that a timeout applied to the entire process. Seeing as we have a convenient cancellation token already, we may as well use it.

Quite likely, the file copy will cancellable by `token` already because it is in `contentStream`.

Reply:
The `DownloadFile` method is also used for other purposes than downloading the install script - it is also used for downloading `nuget.exe` which is around 8-9MB. Not sure whether that is an argument for or against using the cancellation token :-)